src/documentation/sdks/nexmark.md

```diff
@@ -0,0 +1,541 @@
+* **Query8**: Who has entered the system and created an auction in
+  the last period?
+  Illustrates a simple join.
+
+We have augmented the original queries with five more:
+
+* **Query0**: Pass-through.
+  Allows us to measure the monitoring overhead.
+* **Query9**: Winning-bids.
+  A common sub-query shared by **Query4** and **Query6**.
+* **Query10**: Log all events to GCS files.
+  Illustrates windows with large side effects on firing.
+* **Query11**: How many bids did a user make in each session they
+  were active?
+  Illustrates session windows.
+* **Query12**: How many bids does a user make within a fixed
+  processing time limit?
+  Illustrates working in processing time in the Global window, as
+  compared with event time in non-Global windows for all the other
+  queries.
+
+## Benchmark workload configuration
+Here are some of the knobs of the benchmark workload (see [NexmarkConfiguration.java](https://github.com/apache/beam/blob/master/sdks/java/nexmark/src/main/java/org/apache/beam/sdk/nexmark/NexmarkConfiguration.java)).
+
+These configuration items can be passed to the launch command line
+
+### Events generation (defaults)
+* 100 000 events generated
+* 100 generator threads
+* Event rate in SIN curve
+* Initial event rate of 10 000
+* Event rate step of 10 000
+* 100 concurrent auctions
+* 1000 concurrent persons bidding / creating auctions
+
+### Windows (defaults)
+* size 10s
+* sliding period 5s
+* watermark hold for 0s
+
+### Events Proportions (defaults)
+* Hot Auctions = ½
+* Hot Bidders =¼
+* Hot Sellers=¼
+
+### Technical
+* Artificial CPU load
+* Artificial IO load
+
+## Nexmark output
+Here is an example output of the Nexmark benchmark run in streaming mode with the SMOKE suite on the (local) direct runner.
+
+<pre>
+Performance:
+  Conf       Runtime(sec)         Events(/sec)         Results
+  0000                5,5              18138,9          100000
+  0001                4,2              23657,4           92000
+  0002                2,2              45683,0             351
+  0003                3,9              25348,5             444
+  0004                1,6               6207,3              40
+  0005                5,0              20173,5              12
+  0006                0,9              11376,6             401
+  0007              121,4                823,5               1
+  0008                2,5              40273,9            6000
+  0009                0,9              10695,2             298
+  0010                4,0              25025,0               1
+  0011                4,4              22655,2            1919
+  0012                3,5              28208,7            1919
+</pre>
+
+## Benchmark launch configuration
+
+We can specify the Beam runner to use with maven profiles, available profiles are:
+
+    direct-runner
+    spark-runner
+    flink-runner
+    apex-runner
+
+The runner must also be specified like in any other Beam pipeline using
```

Comment:
(throughout this page) if there's a partial sentence before a command line/code block, would be good to add : at the end  e.g. using: